src/components/common/Map/LocationMarker.jsx

```diff
@@ -10,24 +16,88 @@ const markerIcon = new Icon({
   iconSize: [53, 69],
 });
 
+const targetIcon = new Icon({
+  iconUrl: newTarget,
+  iconRetinaUrl: newTarget,
+  popupAnchor: [-0, -0],
+  iconSize: [46, 46],
+});
+
 const LocationMarker = () => {
   const [position, setPosition] = useState(null);
+  const [newTarget, setNewTarget] = useState(null);
 
+  const [, dispatch] = useStore();
   const map = useMap();
+  const history = useHistory();
 
-  useEffect(() => {
+  const findLocation = useCallback(() => {
     map.locate().on('locationfound', function (e) {
       setPosition(e.latlng);
       map.flyTo(e.latlng, map.getZoom());
     });
   }, [map]);
 
+  const handleClick = useCallback(
+    e => {
+      map.on('click', ev => {
+        let latlng = map.mouseEventToLatLng(ev.originalEvent);
+        setNewTarget({ lat: latlng.lat, lng: latlng.lng });
+        dispatch({ type: 'SET_COORDINATES', payload: { lat: latlng.lat, lng: latlng.lng } });
+        history.push(routesPaths.newTarget);
+      });
+    },
+    [map, history, dispatch]
+  );
+
+  const { data: topicsList, isSuccess: topicsSuccess } = useGetTopicsQuery();
+  const { data: targetsList, isSuccess: targetsSuccess } = useGetTargetsQuery();
+
+  const handleIcons = id => {
+    const helper = {};
+
+    for (let i = 0; i < topicsList.topics.length; i++) {
+      helper[topicsList.topics[i].topic.id] = topicsList.topics[i].topic.icon;
+    }
```

Comment:
WDYT about using `forEach` here?
```suggestion
   topicsList.topics.forEach(topic => {
     helper[topic.id] = topic.icon
   })
```

Reply:
I removed this function because it wasn't working.